bootstrap/ssh.go

```diff
@@ -0,0 +1,54 @@
+package bootstrap
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/buildkite/agent/bootstrap/shell"
+	"github.com/pkg/errors"
+)
+
+func sshKeygen(sh *shell.Shell, knownHostsfile, host string) (string, error) {
+	sshKeygen, err := sh.AbsolutePath("ssh-keygen")
+	if err != nil && runtime.GOOS == "windows" {
+		var winErr error
+		if sshKeygen, winErr = findPathToSSHToolsOnWindows(sh); winErr != nil {
+			return "", err
+		}
+	}
+
+	return sh.RunAndCapture(sshKeygen, "-f", knownHostsfile, "-F", host)
```

Comment:
We need to capture stdout and stderr separately — only stdout should be appended to the known hosts file.